### DISCUSSION TITLE
HMIP SWO PL sunshineduration overflow and negative duration

### DISCUSSION THREAD

**Discussion by l-mb** [SukramJ/aiohomematic#1495] [Q&A]
### The problem

Hi, I have a HMIP SWO PL weather station, and that also exposes sensor.hmip_swo_pl_XXXXXX_sunshineduration_2 which counts the daily minutes of sunshine.

That wraps around at apparently 16xxx (my guess would 16384, a 14 bit field), and then drops to zero.

When using this together with a "rate of change" statistic in HA, there's a day with months of darkness :-)

```
state_class: total
interface_id: ccu3-webui-HmIP-RF
address: 00xxxxxxxxxx:1
model: HmIP-SWO-PL
entity_type: generic
parameter: SUNSHINEDURATION
function: null
value_state: valid
unit_of_measurement: min
friendly_name: Sunshine Duration
```

![Graph of sunshine duration](https://github.com/danielperna84/hahomematic/assets/1162196/e88d331c-1e36-45f1-902e-10a11f6dd98a)

![Change history graph](https://github.com/danielperna84/hahomematic/assets/1162196/e0bbe40d-8a60-440f-8953-9518aaff68a9)

I am uncertain whether the right fix is to have a `total_increasing` instead, or whether the integration should hide this wrap internally to keep going?

**Comment by Baxxy13** [comment]
I had a similar issue with all ENERGY_COUNTER's (PSM/FSM/BSM) since Version 1.55.0 (2024-02-02).
(related change: "Use state_class total instead of total_increasing for sensors")

Discussed here: https://github.com/danielperna84/hahomematic/discussions/1470

As a result i customized all my "COUNTER's" back to "total_increasing" and it's working now without problems like before 1.55.0.

**Comment by SukramJ** [comment]
Please use HA [customize](https://www.home-assistant.io/docs/configuration/customizing-devices/#customizing-entities) to change the state_class to total_increasing, and report back if it works.

**Reply by l-mb** [reply]
I've made the change, but will only see whether it works in a few days once it should wrap again. Will report back.

**Reply by l-mb** [reply]
Yes, this seems to have handled the wrap correctly. 

Seems the correct choice for this one and probably the `rain_counter` as well.

**Reply by SukramJ** [reply]
Thanks for your feedback.
Version 1.59.0 already contains these changes.
